sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/MethodsShouldUseBaseTypes.cs

```diff
@@ -124,45 +124,43 @@ private SyntaxNode GetNextUnparenthesizedParent(SyntaxNode node)
             return node;
         }
 
-        private ITypeSymbol FindParameterUseAsType(IdentifierNameSyntax identifier, SemanticModel semanticModel)
+        private static ITypeSymbol FindParameterUseAsType(IdentifierNameSyntax identifier, SemanticModel semanticModel)
         {
+            var callSite = semanticModel.GetEnclosingSymbol(identifier.SpanStart)?.ContainingAssembly;
+
             var identifierParent = GetNextUnparenthesizedParent(identifier);
 
             if (identifierParent is ConditionalAccessExpressionSyntax conditionalAccess)
             {
                 if (conditionalAccess.WhenNotNull is MemberBindingExpressionSyntax binding)
                 {
-                    var name = binding.Name;
-                    if (name == null)
-                    {
-                        return null;
-                    }
-
-                    return HandlePropertyOrField(identifier, semanticModel.GetSymbolInfo(name).Symbol);
+                    return binding.Name == null

```

Comment:
[minor] I find it more readable with the opposite condition (i.e. when `null` is the else part).